### PR TITLE
VACMS-14000: Adds and populates meta tags on VBA Facility

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #14000 

## Testing done
Locally

## Screenshots
Screenshot 2023-12-13 at 4 14 01 PM

### In the browser
As an admin
- [x] Go to [/admin/content](https://pr16395-bfqjp9xc1t3u8qbjj1xqcyatlwlw1nnq.ci.cms.va.gov/admin/content?title=&type=vba_facility&moderation_state=All&owner=All), filtered for VBA Facility
- [x] Confirm that there are 482 nodes
- [x] Go to [Wilmington VA Regional Benefit Office](https://pr16395-bfqjp9xc1t3u8qbjj1xqcyatlwlw1nnq.ci.cms.va.gov/wilmington-va-regional-benefit-office/locations/wilmington-va-regional-benefit-office)
- Use the developer tools to check the page code
- [x] Confirm that you do **not** see a `<meta name="description">` element
- [x] Confirm that you do **not** see a `<meta name="og:description">` element

### In the [terminal](https://tugboat.vfs.va.gov/tush/6578a479ad79a291083b5a79)
- [x] Run the following: `drush scr ./scripts/content/VACMS-14000-vba-facility-meta-tags.php`
- [x] Confirm that the terminal output describes that the number nodes processed matches the search results from `/admin/content`
- [x] Flush the cache: `drush cr`

### In the browser
As an admin
- [x] Check the [Wilmington VA Regional Benefit Office](https://pr16395-bfqjp9xc1t3u8qbjj1xqcyatlwlw1nnq.ci.cms.va.gov/wilmington-va-regional-benefit-office/locations/wilmington-va-regional-benefit-office)
- [x] Use developer tools to check the code
- [x] Confirm that you **do** see a `<meta name="description">` element, and that it says "Placeholder text: Veterans benefits offices provide services to Veterans ... "
- [x] Confirm that you **do** see a `<meta name="og:description">` element and that it says "Placeholder text: Veterans benefits offices provide services to Veterans ... "
- [x] Check [Pre-Discharge Site at Bruce W. Carter Department of Veterans Affairs Medical Center](https://pr16395-bfqjp9xc1t3u8qbjj1xqcyatlwlw1nnq.ci.cms.va.gov/st-petersburg-va-regional-benefit-office/locations/pre-discharge-site-at-bruce-w-carter-department-of-veterans-affairs-medical-center)
- [x] Confirm that you **do** see a `<meta name="description">` element, and that it says "Placeholder text: Veterans benefits offices provide services to Veterans ... "
- [x] Confirm that you **do** see a `<meta name="og:description">` element and that it says "Placeholder text: Veterans benefits offices provide services to Veterans ... "
- [x] Visit the [Recent log messages](https://pr16395-bfqjp9xc1t3u8qbjj1xqcyatlwlw1nnq.ci.cms.va.gov/admin/reports/dblog)
  - [x] Confirm that messages in the log correspond with the outcome of the script 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

File: scripts/content/VACMS-14000-vba-facility-meta-tags.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+/**
+ * Setting the VBA facility meta tags.
+ *
+ * @param array $sandbox
+ *   Modeling the structure of hook_update_n sandbox.
+ *
+ * @return string
+ *   Status message.
+ */
+function va_gov_set_vba_facility_meta_tags(array &$sandbox) {
```
Comment (swirtSJW) [inline scripts/content/VACMS-14000-vba-facility-meta-tags.php:33]: Minor and not worth moving.  Consider for next time around that this could have been done as a "[post deploy hook](https://pasankg.medium.com/using-drupal-drush-deploy-hook-2d2acd38d237)" so that it would run itself without needing a manual step.

File: scripts/content/VACMS-14000-vba-facility-meta-tags.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+/**
+ * Update the node__field_cc_meta_tags table to set the entity fetch field.
+ *
+ * @param string $nodeId
+ *   The id of the node.
+ * @param string $revisionId
+ *   The id of the node revision.
+ */
+function update_field_cc_meta_tags_table(string $nodeId, string $revisionId) {
```
Comment (swirtSJW) [inline scripts/content/VACMS-14000-vba-facility-meta-tags.php:115]: Minor and not worth changing.  Consider abstracting this function one level deeper where it is not specific to meta tags but is generic to updating a fetch field.  It would need to add a bundle parameter. Then move it to the script library.

**Review by swirtSJW — COMMENTED** [review]
I have not tested this yet, but will later tonight.

File: config/sync/core.entity_form_display.node.vba_facility.default.yml [hunk 3]
```diff
@@ -36,7 +37,7 @@ dependencies:
     - field.field.node.vba_facility.field_table_of_contents
     - field.field.node.vba_facility.field_timezone
     - node.type.vba_facility
-    - workflows.workflow.editorial
+    - workflows.workflow.restricted_archive
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.vba_facility.default.yml:40]: Seems odd this is changing this now.  This is unrelated to this PR isn't it?

Comment (omahane) [reply config/sync/core.entity_form_display.node.vba_facility.default.yml:40]: Yes, it is unrelated. I also thought it odd. But it's correct, and this vestigial workflow should be corrected from main. Are you suggesting we pull this change out and do it in a separate PR that is specific to workflow?

File: scripts/content/VACMS-14000-vba-facility-meta-tags.php (hunk 1 above)
Comment (omahane) [reply scripts/content/VACMS-14000-vba-facility-meta-tags.php:33]: Thanks for the suggestion. I'll look into that at the next opportunity.

File: scripts/content/VACMS-14000-vba-facility-meta-tags.php (hunk 2 above)
Comment (omahane) [reply scripts/content/VACMS-14000-vba-facility-meta-tags.php:115]: Yes, good thought. I was thinking that this is something that is going to likely need to happen again.

File: config/sync/core.entity_form_display.node.vba_facility.default.yml (hunk 3 above)
Comment (swirtSJW) [reply config/sync/core.entity_form_display.node.vba_facility.default.yml:40]: No.  Mainly just wanted to note it as odd... and thought maybe you could shed light on it.

Comment (omahane) [reply config/sync/core.entity_form_display.node.vba_facility.default.yml:40]: It is odd, so I ran a little test on Service Region. It turns out the dependency of the manage form display config is only picked up when updating the form itself. Changing the workflow does not result in a config change for the form. So I'll update the Service Region config, too, for consistency.

**Review by swirtSJW — APPROVED** [review]
Confirmed all the QA steps
image

image

image

image

Pattern is set using centralized content.
image

**Review by edmund-dunn — APPROVED** [review]
i see no issues here.
